### PR TITLE
Feature ongoing game

### PR DESCRIPTION
**Pull request by paulSanchezMarmelab**
[feature_ongoing_game.webm](https://user-images.githubusercontent.com/130989826/232828855-5dc7a588-1957-424a-a31d-56786164c886.webm)

- [X] added playedBy support
- [X] added lastPlayed support

### CODE REVIEW

**Review by fzaninotto — COMMENTED** [review]
Your code should really separate the game model from the game representation. All your files mix both intents. This will be very hard to maintain.

**Comment by paulSanchezMarmelab** [comment]
> Your code should really separate the game model from the game representation. All your files mix both intents. This will be very hard to maintain.

I'm not quite sure to see exactly where the issue is. Did you mean that "Tile" should only have render-oriented properties and that logic tied to game states like "playedBy" and "isLastPlayed" should be in a distinct part of the code ?

**Comment by slax57** [comment]
I forgot to mention this after I reviewed https://github.com/marmelab/kamon/pull/4, but we should have documented how to run the CLI with the `-f` parameter.

For this time only I don't recommend adding a `makefile` target for this, but instead I would recommend using `yarn` directly.

The command should be something like 

```sh
yarn start -f my_save_file.json
```

Can you add it to the `Readme` on this PR please?

**Comment by paulSanchezMarmelab** [comment]
> I forgot to mention this after I reviewed #4, but we should have documented how to run the CLI with the `-f` parameter.
> 
> For this time only I don't recommend adding a `makefile` target for this, but instead I would recommend using `yarn` directly.
> 
> The command should be something like
> 
> ```shell
> yarn start -f my_save_file.json
> ```
> 
> Can you add it to the `Readme` on this PR please?

I added instructions to Readme file.

File: Readme.md [hunk 1]
```diff
@@ -8,14 +8,19 @@ Play the game of Kamon on the console
...
 ```
 
 ### Playing
+
 ```sh
+## make a game from save.json
 make run
+
+## make a game from a target save file
+yarn start -f my_save_file.json
```
Comment (slax57) [inline Readme.md:23]:
```suggestion
## initialize a new game
make run

## resume a game from a target save file
yarn start -f my_save_file.json
```

File: src/tile/tile.ts [hunk 2]
```diff
@@ -10,19 +10,25 @@ export const colors = [
 ] as const;
 export const symbols = ["A", "B", "C", "D", "E", "F"] as const;
 
-export type Color = typeof colors[number];
-export type Symbol = typeof symbols[number];
+export type Color = (typeof colors)[number];
+export type Symbols = (typeof symbols)[number];
```
Comment (slax57) [inline src/tile/tile.ts:14]:
```suggestion
export type Symbol = (typeof symbols)[number];
```

File: src/tile/tile.ts [hunk 3]
```diff
@@ -10,19 +10,25 @@ export const colors = [
...
 
 export type Tile = PlayableTile | NeutralTile;
 
+type Players = "black" | "white";
```
Comment (slax57) [inline src/tile/tile.ts:18]:
```suggestion
type Player = "black" | "white";
```

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by slax57 — APPROVED** [review]

File: src/tile/tile.ts [hunk 4]
```diff
@@ -44,11 +50,28 @@ export const getAllSymbols = (): Tile[] => {
...
+    if (tile.playedBy === "white") {
+      dynamicChalk = dynamicChalk.bgWhite;
+    }
+  }
+
+  if (tile.lastPlayed === true) {
+    dynamicChalk = dynamicChalk.bold.underline;
+  }
+
+  return dynamicChalk(tile.symbol);
```
Comment (slax57) [inline src/tile/tile.ts:76]:
```suggestion
  return dynamicChalk(`${tile.symbol} `);
```
Otherwise this is a regression

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by slax57 — APPROVED** [review]

**Pull request merged**